### ISSUE TITLE
1.77 Errors in set program.

### ISSUE THREAD

**Issue by Wojbie** [dan200/ComputerCraft#85]
Duplicated error report from : http://www.computercraft.info/forums2/index.php?/topic/25682-mc-189cc-177-settings-program-bug/
Due to 2 small errors in set program it will always save even if data was not changed because it gets invalid data and it will unset value instead of setting it to false because of wrong test used.
Problem lies in lines 31-32
They are:
```lua
local oldValue = settings.get( sValue )
if value then
```
And should be
```lua
local oldValue = settings.get( sName )
if value ~= nil then
```

**Comment by dan200** [comment]
Grr. Wish this had been reported during beta testing.
Will be fixed in 1.77

**Comment by Wojbie** [comment]
I admit to feeling embarrassed about this one. I was playing with it quite lot and it missed me too.
Also 2nd thing i just noticed:
In line 6 of set program you are using print instead of textutils.pagedPrint. That means if there are more settings then screen lines only last one will be visible? Not sure if bug just pointing that out.
Also does fixed in 1.77 mean you are doing ghost fix? :D

**Comment by dan200** [comment]
Means a typo!
